### PR TITLE
VZ-6420: Add test to ensure CRDs are not deleted after uninstall

### PR DESCRIPTION
**Pull request by bobdonat**
This pull request adds an acceptance test to verify that CRDs are not deleted after an uninstall.  This test is run during the uninstall resiliency AT.

### CODE REVIEW

File: tests/e2e/uninstall/verifycrds/verify_crds_test.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+	. "github.com/onsi/ginkgo/v2"
+	"github.com/verrazzano/verrazzano/pkg/test/framework"
+	"github.com/verrazzano/verrazzano/tests/e2e/pkg"
+	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"strings"
+)
+
+// Expected verrazzano.io CRDs after uninstall
+var verrazzanoiocrds = map[string]bool{
+	"ingresstraits.oam.verrazzano.io":                              false,
```
Comment (davisgomes) [inline tests/e2e/uninstall/verifycrds/verify_crds_test.go:17]: Are any of the values ever true? Do we need the map value?

File: tests/e2e/uninstall/verifycrds/verify_crds_test.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+	})
+})
+
+// checkCRds checks for both expected CRDs and unexpected CRDs for a given CRDs suffix (for example, verrazzano.io)
+func checkCrds(crds *apiextv1.CustomResourceDefinitionList, expectdCrds map[string]bool, suffix string) {
+	unexpectedCrd := false
+	for _, crd := range crds.Items {
+		_, ok := expectdCrds[crd.Name]
+		if ok {
+			expectdCrds[crd.Name] = true
```
Comment (davisgomes) [inline tests/e2e/uninstall/verifycrds/verify_crds_test.go:174]: Would it be easier to append this value to an unexpected found and loop through that later? I get why you implemented the map for O(1) lookup time. Would a "Set" (`map[string]struct{}`) with a  work better? O(1) lookup time, but the struct takes up zero bytes. Not that big of a deal if your implementation is easier.

Comment (davisgomes) [reply tests/e2e/uninstall/verifycrds/verify_crds_test.go:174]: the format of the map would be
```
var istioiocrds = map[string]bool{
	"authorizationpolicies.security.istio.io":  {},
        ...
}
```

File: tests/e2e/uninstall/verifycrds/verify_crds_test.go (hunk 1 above)
Comment (bobdonat) [reply tests/e2e/uninstall/verifycrds/verify_crds_test.go:17]: The value is initialized to false and set to true when a match is found

File: tests/e2e/uninstall/verifycrds/verify_crds_test.go (hunk 2 above)
Comment (bobdonat) [reply tests/e2e/uninstall/verifycrds/verify_crds_test.go:174]: I am not clear on what you are suggesting.

**Review by pfmackin — APPROVED** [review]

**Review by davisgomes — APPROVED** [review]

**Pull request merged**